src/Altinn.Apps/AppFrontend/react/altinn-app-frontend/src/features/form/rules/fetch/fetchRulesSagas.ts

```diff
@@ -1,17 +1,42 @@
 import { SagaIterator } from 'redux-saga';
-import { call, takeLatest } from 'redux-saga/effects';
+import { call, takeLatest, select } from 'redux-saga/effects';
+import { IAltinnWindow, IInstance } from 'altinn-shared/types';
+import { IApplicationMetadata } from 'src/shared/resources/applicationMetadata';
 import { get } from '../../../../utils/networking';
 import { getRuleModelFields } from '../../../../utils/rules';
 import Actions from '../rulesActions';
 import * as FetchActions from './fetchRulesActions';
 import * as ActionTypes from '../rulesActionTypes';
 import QueueActions from '../../../../shared/resources/queue/queueActions';
+import { IRuntimeState, ILayoutSets } from '../../../../types';
+import { getLayouytsetForDataElement } from '../../../../utils/layout';
+import { getDataTaskDataTypeId } from '../../../../utils/appMetadata';
 
+const layoutSetsSelector = (state: IRuntimeState) => state.formLayout.layoutset;
+const instanceSelector = (state: IRuntimeState) => state.instanceData.instance;
+const applicationMetadataSelector = (state: IRuntimeState) => state.applicationMetadata.applicationMetadata;
+
+/**
+ * Saga to retrive the rule configuration defining which rules to run for a given UI
+ */
 function* fetchRuleModelSaga({
   url,
 }: FetchActions.IFetchRuleModel): SagaIterator {
   try {
-    const ruleModel = yield call(get, url);
+    const { org, app } = window as Window as IAltinnWindow;
+    const layoutSets: ILayoutSets = yield select(layoutSetsSelector);
+    const instance: IInstance = yield select(instanceSelector);
+    const aplicationMetadataState: IApplicationMetadata = yield select(applicationMetadataSelector);
+    const dataType: string =
+    getDataTaskDataTypeId(instance.process.currentTask.elementId, aplicationMetadataState.dataTypes);
+    let apiUrl: string = url;
+
+    if (layoutSets != null) {
+      const layoutSetId: string = getLayouytsetForDataElement(instance, dataType, layoutSets);
+      apiUrl = `${window.location.origin}/${org}/${app}/api/rulehandler/${layoutSetId}`;
```

Comment:
Same comment as before. Can you add the url to the urlHelper, with a layoutSetId parameter? And then remove the url parameter from the action call.

Reply:
done